plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/imports/ReferenceResolutionFinder.java

```diff
@@ -149,8 +151,10 @@ public void findResolutions(
 		boolean needCollisionCheck = !isUnresolvedReference && !isPropertyAccess;
 
 		List<IEObjectDescription> candidates = new ArrayList<>(512);
+		Map<IEObjectDescription, N4JSProjectConfigSnapshot> projects = new HashMap<>();
 		Set<QualifiedName> candidateNames = needCollisionCheck ? new HashSet<>() : null;
-		collectAllElements(scope, candidates, candidateNames, acceptor);
+		Set<QualifiedName> collisioningModules = new HashSet<>();
+		collectAllElements(scope, wc, candidates, projects, candidateNames, collisioningModules, acceptor);
```

Comment:
Did you check the runtime of this invocation in the OPR workspace before/after your changes?